### PR TITLE
Files in format

### PR DESCRIPTION
**Pull request by KushanRaj**

### CODE REVIEW

File: datasets/bevdetection.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from torch.utils.data import Dataset
+import numpy as np
+import os 
+from glob import glob
+from transformation.Frame import Frame
+import torch
+from transformation.BEV import BEV
+
+class BEVDetection(Dataset):
```
Comment (MukundVarmaT) [inline datasets/bevdetection.py:9]: change this to KittiObject

File: datasets/bevdetection.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        self._label_path = []
+        self._calib_path = []
+        self._image_path = []
+        
+        self._scan_paths.extend(sorted(glob(os.path.join(self.root,split,"velodyne", '*.bin'))))
+        self._label_path.extend(sorted(glob(os.path.join(self.root,split,"label_2", '*.txt'))))
+        self._calib_path.extend(sorted(glob(os.path.join(self.root,split,"calib", '*.txt'))))
+        self._image_path.extend(sorted(glob(os.path.join(self.root,split,"image_2", '*.png'))))
+        
+        if len(self._scan_paths) != len(self._label_path):
```
Comment (MukundVarmaT) [inline datasets/bevdetection.py:38]: this will be valid only for train and validation set so add an if statement here

File: datasets/bevdetection.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            raise ValueError((f"number of scans {len(self._scan_paths)} not equal to number of labels {len(self._label_path)}"))     
+  
+    
+    @staticmethod
+    def get_bevfrom_scan(file_name,imgsize):
```
Comment (MukundVarmaT) [inline datasets/bevdetection.py:43]: i think you must just do get scan and the bev conversion is done in the __getitem__ method anyways

File: datasets/bevdetection.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        return BEV(imgsize).transform_to_BEV(scan)
+    
+    @staticmethod
+    def get_2D_label(lfile,cfile,imgsize):
+        """
+        open label file - [semantic label(first 16 bits), instance label(last 16 bits)]
+        returns semantic label, instance label
+        """
+        
+        label,points = Frame(lfile,cfile,imgsize)
```
Comment (MukundVarmaT) [inline datasets/bevdetection.py:60]: I feel that the classes in Frame can be added to this file itself as its relevant to the dataset
Also i am a bit unsure about why you need to create an object of a class for every file (i dont know if its efficient as well). For now we will keep it like this, no problem but we need to think about whether its required

File: datasets/bevdetection.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        return label,points
+
+    def __getitem__(self, idx):
+        scan = self.get_bevfrom_scan(self._scan_paths[idx],self.IMG_SIZE)
```
Comment (MukundVarmaT) [inline datasets/bevdetection.py:64]: this will be using the transforms object of the class.

File: transformation/BEV.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
+#imports
+import numpy as np
+
+import os
+#function to convert PointCloud to BEV
+class BEV():
+
+
+    def __init__(self,cfg):
```
Comment (MukundVarmaT) [inline transformation/BEV.py:9]: add a `__call__` method here. this should essentially do all the transforms if i call this transformation from anywhere
